### PR TITLE
More multi-appdomain support removal

### PR DESCRIPTION
**Pull request by davidwrighton**
- Remove DomainLocalBlock
- Remove DomainModule
- Remove FindDomain(File/Module/Assembly)
- Remove GetDomainModule
- Remove variants of GetDomain(File/Assembly) that take AppDomain parameter

### CODE REVIEW

File: src/debug/daccess/request.cpp [hunk 1]
```diff
@@ -3226,8 +3225,22 @@ ClrDataAccess::GetDomainLocalModuleDataFromAppDomain(CLRDATA_ADDRESS appDomainAd
...
+    DomainLocalModule* pLocalModule = NULL;
+
+    AppDomain::AssemblyIterator i = pAppDomain->IterateAssembliesEx(
+        (AssemblyIterationFlags)(kIncludeLoading | kIncludeLoaded | kIncludeExecution));
+    CollectibleAssemblyHolder<DomainAssembly *> pDomainAssembly;
+    while (i.Next(pDomainAssembly.This()))
+    {
+        if (pDomainAssembly->GetModule()->GetModuleIndex() == index)
+        {
+            pLocalModule = pDomainAssembly->GetModule()->GetDomainLocalModule();
```
Comment (jkotas) [inline src/debug/daccess/request.cpp]: I do not think that this will ever find anything. Should the whole method just return `E_INVALIDARG`?

Comment (davidwrighton) [reply src/debug/daccess/request.cpp]: Hmm, I was under the impression we still had Module indices. (I'm not sure why we would still have them, but I thought they were there.) In that case, I believe we could still find these things. I'll do some investigation.

Comment (janvorli) [reply src/debug/daccess/request.cpp]: The module index is still used in `JIT_GetSharedNonGCThreadStaticBase` and `JIT_GetSharedGCThreadStaticBase` to find ThreadLocalModule using `ThreadStatics::GetTLMIfExists(index)`

Comment (jkotas) [reply src/debug/daccess/request.cpp]: We do have module indexes, but we do not have module IDs. The code the calls `Module::IDToIndex(moduleID)` - like this method - should be unreachable.

Comment (davidwrighton) [reply src/debug/daccess/request.cpp]: Sigh, right. I was ripping out these very bits of logic, and somehow missed this. Sigh, Well, I'll pull a bit more of those details out, as it was still confusing to me.

Comment (davidwrighton) [reply src/debug/daccess/request.cpp]: And yes, since the moduleid here is an int, it will never match anything. E_INVALIDARG is correct.

**Pull request merged**